cadl-extension/src/main/java/com/azure/cadl/mapper/CadlPrimitiveMapper.java

```diff
@@ -21,6 +21,8 @@ public static PrimitiveMapper getInstance() {
     protected IType createPrimitiveType(PrimitiveSchema primaryType) {
         if (primaryType.getType() == Schema.AllSchemaTypes.DATE) {
             return ClassType.LocalDate;
+        } else if (primaryType.getType() == Schema.AllSchemaTypes.BINARY){
+            return ClassType.BinaryData;
```

Comment:
`Schema.AllSchemaTypes.BINARY` seems not processed in `PrimitiveMapper`. How does it handle binary date from m4?

Reply:
This is M4 example, it has `requestMediaTypes` on each operation, `requestMediaTypes` will map different content types to different input schema.
![image](https://user-images.githubusercontent.com/87355844/224904679-5033cf3d-fa86-4a82-ae03-7d26e9e85a66.png)
![image](https://user-images.githubusercontent.com/87355844/224904745-02f414d7-cc04-4419-bc65-55ddf5b21798.png)
![image](https://user-images.githubusercontent.com/87355844/224908659-a83cc29f-5d2d-4e9b-85f3-f34a62b1d47c.png)

It is processed here: https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/mapper/MethodGroupMapper.java#L89

Reply:
How it process that, I cannot find code `requestMediaTypes` in the whole project?

Reply:
E.g. in vanilla mediatypes test, it has
```java
    @ServiceMethod(returns = ReturnType.SINGLE)
    public String analyzeBody(ContentType contentType, Flux<ByteBuffer> input, Long contentLength) {
        return analyzeBodyWithResponse(contentType, input, contentLength, Context.NONE).getValue();
    }
```

Ignore the `contentLength` (DPG does not have it), it has the `Flux<ByteBuffer> input`. I assumed something mapped to `Flux<ByteBuffer>`?

Reply:
Also, we are talking about single content-type with binary upload.

We can avoid discussion of multiple content-type for now.

Reply:
> ![image](https://user-images.githubusercontent.com/87355844/224911299-61801a2c-f0e1-4e0e-9dd6-c7da9cb8c2c5.png) requests refer to the item in requestMediaTypes, we proccess requests.

I see. But still this question. How vanilla process this without mapping in `PrimitiveMapper`? Your code is DPG only (which map any payload to BinaryData)

Reply:
This is where we map `BinarySchema` to `FluxByteBuffer`
https://github.com/Azure/autorest.java/blob/main/javagen/src/main/java/com/azure/autorest/mapper/BinaryMapper.java#L28

call stack:
![image](https://user-images.githubusercontent.com/87355844/224930534-d6d1313d-057e-4db1-8ba2-43e68c9e8945.png)

Reply:
So this change here is not needed?

Reply:
I think so..I'm testing..